docs/upgrade.md

```diff
@@ -89,6 +89,231 @@ process, for example:
     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
     ```
 
+# Upgrading to v1.61.0
+
+## New signature for the spam checker callback `user_may_join_room`
```

Comment:
I feel like all these sections can be merged together since they're changes are essentially the same. I'd rework this by listing the callbacks that have been changed and explaining the changes like you do here (except just once).